app/models/checkout.rb

```diff
@@ -1,34 +1,43 @@
 class Checkout < ApplicationRecord
   CHECKOUT_PERIOD_IN_DAYS = 14
-  FINE_AMOUNT_PER_DAY_IN_CENTS = 0.10
+  FINE_AMOUNT_PER_DAY_IN_CENTS = 10
 
   belongs_to :user
   belongs_to :book
 
   validates :user, presence: true
-  validates :book, presence: true, uniqueness: true
+  validates :book, presence: true #, uniqueness: true
+  validates :due_on, presence: true
 
-  def due_date
-    self.created_at + CHECKOUT_PERIOD_IN_DAYS.days
-  end
+  monetize :fine_cents
 
-  def fine
```

Comment:
I would probably create a `CalculateFine` class that does this. I imagine it would be used a bit like this:

```ruby
CalculateFine.for(checkout)
CalculateFine.including_related_checkout_for(checkout)
```